examples/digital_fingerprinting/production/morpheus/dfp/stages/multi_file_source.py

```diff
@@ -109,17 +117,17 @@ def _polling_generate_frames_fsspec(self) -> typing.Iterable[fsspec.core.OpenFil
             # need to re-ingest that new file.
             files_seen = file_set
 
-            pre_yield_time = time.time()
             if len(filtered_files) > 0:
                 yield fsspec.core.OpenFiles(filtered_files, fs=files.fs)
 
-            post_yield_time = time.time()
+            curr_time = time.monotonic()
 
-            # because yielding to the output channel can block, we should only sleep when yielding didn't take longer
-            # than the watch interval
-            yield_time = post_yield_time - pre_yield_time
-            if yield_time < self._watch_interval:
-                time.sleep(self._watch_interval - yield_time)
+            # If we spent more than `self._watch_interval` doing work and/or yielding to the output channel blocked,
+            # then we should only sleep for the remaining time until the next update epoch.
+            sleep_duration = next_update_epoch - curr_time
+            if (sleep_duration > 0):
+                time.sleep(sleep_duration)
+                curr_time = time.monotonic()
```

Comment:
```suggestion
            curr_time = time.monotonic()
```

This should be unindented, right?